### PR TITLE
Skip fleet API calls on integration tests

### PR DESCRIPTION
**Pull request by amirbenun**
### Summary of your changes
The `test_sanity_checks` is called on both flows:
1. Integration tests
2. Sanity tests

On the integration tests, fleet and kibana are not up and therefore we can't use their API.
This fix skip calling fleet API when the `KIBANA_URL` environment variable is missing.

### Checklist
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added the necessary README/documentation (if appropriate)

#### Introducing a new rule?

- [ ] Generate rule metadata using [this script](https://github.com/elastic/cloudbeat/tree/main/security-policies/dev#generate-rules-metadata)
- [ ] Add relevant unit tests
- [ ] Generate relevant rule templates using [this script](https://github.com/elastic/cloudbeat/tree/main/security-policies/dev#generate-rule-templates), and open a PR in [elastic/packages/cloud_security_posture](https://github.com/elastic/integrations/tree/main/packages/cloud_security_posture)

### CODE REVIEW

File: tests/commonlib/agents_map.py [hunk 1]
```diff
@@ -53,6 +53,7 @@ def load_map(self):
         """
         Load the components map with the agent IDs.
         """
+        logger.info("amiramir loading map")
```
Comment (oren-zohar) [inline tests/commonlib/agents_map.py:56]: Please remove

File: tests/integration/tests/test_sanity_checks.py [hunk 2]
```diff
@@ -274,6 +274,12 @@ def wait_components_list(actual: AgentComponentMapping, expected: AgentExpectedM
         list: The list of agents running the specified component.
     """
 
+    # Skip waiting for agents if fleet is not available.
+    if not elasticsearch.kibana_url:
+        logger.info("amiramir skip loading map")
```
Comment (oren-zohar) [inline tests/integration/tests/test_sanity_checks.py:279]: Please remove

**Review by oren-zohar — CHANGES_REQUESTED** [review]

**Comment by amirbenun** [comment]
A successfull run: https://github.com/elastic/cloudbeat/actions/runs/7686877924

File: tests/conftest.py [hunk 3]
```diff
@@ -298,5 +298,4 @@ def agents_actual_components() -> AgentComponentMapping:
     @return: an mapping of the agent components.
     """
     mapping = AgentComponentMapping()
-    mapping.load_map()
     return mapping
```
Comment (gurevichdmitry) [inline tests/conftest.py:301]: can be replaced by:

`return AgentComponentMapping()`

**Review by gurevichdmitry — COMMENTED** [review]
@amirbenun The successful run is only for sanity checks. Could you please execute the original workflow to ensure that this fixes both integrations and sanity checks together?

btw, another option to fix that is to execute only integration tests by eliminating sanity checks using a regular expression pattern.
`poetry run pytest -k "azure and not cspm_azure" --alluredir=./allure/results/ --clean-alluredir`

**Review by oren-zohar — APPROVED** [review]

**Comment by amirbenun** [comment]
Another successful run: https://github.com/elastic/cloudbeat/actions/runs/7693498240/job/20962379730

**Pull request merged**